### PR TITLE
CLI surveys

### PR DESCRIPTION
**Pull request by ismael-mendoza**
This PR extends the CLI functionality to allow for completely customizable surveys (including PSF) from the command line. 

Users can also create their own survey config files which might be easier. I will add some documentation for this soon

### CODE REVIEW

**Comment by ismael-mendoza** [comment]
This PR is basically ready to merge, I just need to update the corresponding notebooks and tutorials. Reviews are appreciated! @thuiop what do you think? 

Basically I made all the survey objects config files in `conf/surveys` so for example Rubin has a config object `conf/surveys/Rubin.yaml`. The purpose is so that users can modify the config object on-the-fly from the CLI and create their surveys in that way. For example, if you want to run `btk` from end-to-end with the Rubin survey with a slightly different airmass  you would do:

```
btk surveys=Rubin surveys.Rubin.airmass=1.1 (...)
```

You can also do multiple surveys! (although won't work until #174 is merged): 

```
btk surveys=[Rubin, HST] surveys.Rubin.airmass=1.1 surveys.HST.airmass=0.9 (...)
```

All survey objects were removed from `btk/survey.py` (hopefully this looks cleaner as well). You can still access them via the `btk.survey.get_surveys` function and in that way get the objects we were using before so we didn't really lose anything. For example: 

```
# get Rubin btk.survey.Survey object
from btk.survey import get_surveys
Rubin = get_surveys('Rubin')

# get HST and Rubin Survey objects
L = get_surveys(['Rubin', 'HST'])
Rubin = L[0]
HST = L[1]
```
Hopefully this make sense, there are notes also in  help [file](https://github.com/LSSTDESC/BlendingToolKit/blob/b361bffa44e8ad6240e1492e3d778461df64f0fe/conf/hydra/help/btk_help.yaml#L47) and I will add to documentation in upcoming PR.

**Review by thuiop — DISMISSED** [review]
This review seems nice to me, I only have a few minor comments ; I think storing the surveys in config files is better than in a python file.

File: conf/surveys/Rubin.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# https://www.lsst.org/about/camera/features
```
[the remark this answers is not in the document]
Comment (ismael-mendoza) [reply conf/surveys/Rubin.yaml:1]: oh yes, let's change it, The other one is definitely more standard.

File: tests/test_error_cases.py [hunk 2]
```diff
@@ -1,9 +1,7 @@
 import pytest
 from conftest import data_dir
 
-import btk
-from btk.survey import Rubin
-
+import btk.survey
```
[the remark this answers is not in the document]
Comment (ismael-mendoza) [reply tests/test_error_cases.py:4]: Good point, I did a bit of cleaning and only import the necessary things in each test (this way we know what features each test is using at a glance).

File: conf/hydra/help/btk_help.yaml [hunk 3]
```diff
@@ -24,13 +22,13 @@ template: |-
...
-  and saving the corresponding metrics.
+  and saving the corresponding metrics via a MetricsGenerator.
 
   Assuming that BTK has been pip installed, you can run btk from the command line like e.g.
 
   btk sampling=default draw_blends=catsim max_number=3 save_path=/directory/to/save/results cpus=1
-    verbose=False surveys=[Rubin, HST] sampling=random catalog.name=catsim
-    use_metrics=['detection', 'segmentation'] (...)
+    verbose=False surveys=[Rubin, HST] surveys.Rubin.airmass=1.1
+    sampling=random catalog.name=catsim use_metrics=['detection', 'segmentation'] (...)
```
[the remark this answers is not in the document]
Comment (ismael-mendoza) [reply conf/hydra/help/btk_help.yaml:31]: it's a typo actually should be 'default' which refers to the DefaultSamplingFunction , thanks!

**Comment by ismael-mendoza** [comment]
Actually, I will merge this for now and revisit the notebooks at the same time as the tutorials.

**Pull request merged**